### PR TITLE
ESLint: Node 8+ WARNINGs --> ERRORs

### PR DESCRIPTION
**Pull request by ace-n**
I tried to break this up piece-wise by commit - so if a sample you own is affected, feel free to review  only the relevant commit(s).

### CODE REVIEW

File: functions/sendgrid/index.js [hunk 1]
```diff
@@ -287,13 +283,12 @@ exports.sendgridWebhook = (req, res) => {
...
-function getTable() {
-  const dataset = bigquery.dataset(config.DATASET);
+const getTable = async () => {
+  let dataset = bigquery.dataset(config.DATASET);
 
-  return dataset
-    .get({autoCreate: true})
-    .then(([dataset]) => dataset.table(config.TABLE).get({autoCreate: true}));
-}
+  dataset = await dataset.get({autoCreate: true})[0];
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/sendgrid/index.js]: Done - turns out you _can_ destructure without a `const` 🙂

File: functions/spanner/index.js [hunk 2]
```diff
@@ -45,22 +45,20 @@ exports.get = (req, res) => {
...
-      res
-        .status(500)
-        .send(`Error querying Spanner: ${err}`)
-        .end();
+  try {
+    const results = await database.run(query);
+    const rows = results[0].map(row => row.toJSON());
+    rows.forEach(row => {
+      res.write(
+        `SingerId: ${row.SingerId}, AlbumId: ${row.AlbumId}, AlbumTitle: ${row.AlbumTitle}\n`
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/spanner/index.js]: Changed - PTAL?

[the remark this answers is not in the document]
Comment (grant) [reply functions/spanner/index.js]: I'm not sure why we don't just `join` the attributes. We're missing a space between attributes with the current change.

Try changing it again?

File: functions/http/index.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ exports.helloContent = (req, res) => {
   switch (req.get('content-type')) {
     // '{"name":"John"}'
     case 'application/json':
-      name = req.body.name;
+      name = req.body;
```
[the remark this answers is not in the document]
Comment (grant) [reply functions/http/index.js]: Can you comment here and ensure we're assigning `name` to the right value?

File: functions/spanner/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/spanner/index.js]: IMO that's harder to understand at a glance vs. a `+`.

We could also call `res.write()` multiple times, if you think that'd be better.

[the remark this answers is not in the document]
Comment (grant) [reply functions/spanner/index.js]: OK. Either way is fine.

File: functions/scheduleinstance/index.js [hunk 4]
```diff
@@ -31,37 +31,31 @@ const compute = new Compute();
  * @param {!object} callback Cloud Function PubSub callback indicating
  *  completion.
  */
-exports.startInstancePubSub = (event, context, callback) => {
+exports.startInstancePubSub = async (event, context, callback) => {
```
Comment (djmailhot) [inline functions/scheduleinstance/index.js]: I ran this updated code for the tutorial [Scheduling compute instances with Cloud Scheduler](https://cloud.google.com/scheduler/docs/start-and-stop-compute-engine-instances-on-a-schedule)

Actions:
1. Create a GCE instance `dev-instance` and a Cloud Function `startInstancePubSub`.
2. Stop the `dev-instance` instance
3. Run `startInstancePubSub` with a test trigger event targeting `dev-instance`.

Expected:
- Instance is started
- Function returns `Successfully started instance dev-instance`

Actual:
- Instance is started
- Function returns `Function failed on loading user code. Error message: File index.js or function.js that is expected to define function doesn't exist in the root directory.`

Comment (djmailhot) [reply functions/scheduleinstance/index.js]: I re-ran this code.  This time `startInstancePubSub` worked fine, but `stopInstancePubSub` did not.

Actions:
1. Create a GCE instance `dev-instance` and a Cloud Function `stopInstancePubSub`.
3. Run `stopInstancePubSub` with the test trigger event specified in the tutorial.

Expected:
- Instance is stopped
- Function returns `Successfully stopped instance dev-instance`

Actual:
- Instance is stopped
- Function returns `Error: function execution failed. Details:
Cannot read property 'Symbol(Symbol.iterator)' of undefined`

Comment (djmailhot) [reply functions/scheduleinstance/index.js]: Stack trace of the error:
`stopInstancePubSub
gwd5qb1ffhx6
TypeError: Cannot read property 'Symbol(Symbol.iterator)' of undefined at Function.all (<anonymous>) at exports.stopInstancePubSub (/srv/index.js:85:19) at <anonymous> at process._tickDomainCallback (internal/process/next_tick.js:229:7)`

Comment (djmailhot) [reply functions/scheduleinstance/index.js]: Issue resolved

- Instance is stopped
- Function returns `Successfully stopped instance dev-instance`

**Pull request merged**